### PR TITLE
Add unit tests for conditions application and set up CI test

### PR DESCRIPTION
**Pull request by riecke**
- Note that the continuous integration test runs only on the "riecke_tests" branch for now
- The migration scripts for the "conditions" application were completely reset, and need more careful review
- All the backend tests can be run with 
      python manage.py test -p "\*test\*.py"

### CODE REVIEW

File: src/planscape/planscape/settings.py [hunk 1]
```diff
@@ -178,3 +178,8 @@
 ACCOUNT_EMAIL_VERIFICATION = 'none'
 ACCOUNT_LOGOUT_ON_GET = True
 
+# PostGIS values
```
Comment (elsiel23) [inline src/planscape/planscape/settings.py:181]: is there any documentation on where this comes from/how it's used?

File: src/planscape/testing/postgis_setup.sql [hunk 2]
```diff
@@ -0,0 +1,8 @@
+CREATE EXTENSION postgis;
+CREATE EXTENSION postgis_raster;
+ALTER SYSTEM SET postgis.enable_outdb_rasters TO True;
```
Comment (elsiel23) [inline src/planscape/testing/postgis_setup.sql:3]: why are we using outdb rasters here?

File: .github/workflows/django.yaml [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+      matrix:
+        python-version: [3.10.8]
+    
+    services:
+      postgres:
+        image: postgis/postgis:14-3.3
+        env:
+          # must specify password for PG Docker container image
+          # see: https://registry.hub.docker.com/_/postgres?tab=description&page=1&name=10
+          POSTGRES_PASSWORD: password
```
[the remark this answers is not in the document]
Comment (riecke) [reply .github/workflows/django.yaml:24]: A new instance is created for each workflow, by prepending "test_" to the database name.

File: src/planscape/conditions/migrations/0001_initial.py [hunk 4]
```diff
@@ -1,21 +1,21 @@
-# Generated by Django 4.1.1 on 2022-11-02 21:37
+# Generated by Django 4.1.1 on 2022-11-27 19:01
```
[the remark this answers is not in the document]
Comment (riecke) [reply src/planscape/conditions/migrations/0001_initial.py:1]: Django management seems to be a sticky wicket: without it, one has to create the corresponding table "conditions_conditionraster" explicitly in the setup.

We could do that.  But then I was wondering whether we should leave it as a "managed" model.  It gives us some extra commands.  What do you think?

File: src/planscape/conditions/models.py [hunk 5]
```diff
@@ -56,14 +57,16 @@ class ConditionRaster(models.Model):
...
+
     # The name of the raster, which must match the raster_name in the Condition. 
-    # WARNING: raster2pgsql does not work if the field name has any _ chars. 
+    # WARNING: raster2pgsql does not work if this field name has any _ chars. 
     name: models.TextField = models.TextField(null=True)
 
     # A tile in the raster.
-    raster = models.RasterField(null=True)
-
-    class Meta:
```
[the remark this answers is not in the document]
Comment (riecke) [reply src/planscape/conditions/models.py:67]: See comment above.

File: src/planscape/testing/postgis_setup.sql (hunk 2 above)
Comment (riecke) [reply src/planscape/testing/postgis_setup.sql:3]: This is necessary to convert any raster into an external format, e.g., PNG or JPEG for displaying tiles.

File: src/planscape/planscape/settings.py (hunk 1 above)
Comment (riecke) [reply src/planscape/planscape/settings.py:181]: I'll beef this up.

File: .github/workflows/django.yaml (hunk 3 above)
[the remark this answers is not in the document]
Comment (riecke) [reply .github/workflows/django.yaml:24]: Oh! and to answer the question, no, this database is created in GitHub.  It's started from a Docker image, and runs as a separate process on GitHub servers.

File: src/planscape/planscape/settings.py (hunk 1 above)
Comment (riecke) [reply src/planscape/planscape/settings.py]: Done.

File: src/planscape/conditions/tests.py [hunk 6]
```diff
@@ -1,3 +1,71 @@
...
+
+PLANSCAPE_ROOT_DIRECTORY = cast(str, config('PLANSCAPE_ROOT_DIRECTORY'))
+RASTER_TEST_FILE='src/planscape/testing/testdata/random_test.sql'
+
+
+class ConditionTest(TestCase):
+    @classmethod
+    def setUpClass(cls):
+        # Read the test data; this is a GeoTIFF with CRS 9822, ready to load into the
+        # CnditionRaster table.  See testing/create_random_geotiff.py for details.
```
[the remark this answers is not in the document]
Comment (riecke) [reply src/planscape/conditions/tests.py]: Done.

File: .github/workflows/django.yaml (hunk 3 above)
[the remark this answers is not in the document]
Comment (elsiel23) [reply .github/workflows/django.yaml:24]: Thanks for the explanation! :)

File: src/planscape/conditions/migrations/0001_initial.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (elsiel23) [reply src/planscape/conditions/migrations/0001_initial.py:1]: Makes sense, let's keep it managed

File: src/planscape/testing/postgis_setup.sql (hunk 2 above)
Comment (elsiel23) [reply src/planscape/testing/postgis_setup.sql:3]: Ack, thanks

**Review by elsiel23 — APPROVED** [review]

**Pull request merged**